clients/drcachesim/tests/view_test.cpp

```diff
@@ -227,26 +298,28 @@ test_num_memrefs(void *drcontext, instrlist_t &ilist,
 
 bool
 test_skip_memrefs(void *drcontext, instrlist_t &ilist,
-                  const std::vector<memref_t> &memrefs, int skip_memrefs, int num_memrefs)
+                  const std::vector<memref_t> &memrefs,
+                  const std::vector<trace_entry_t> &records, int skip_records,
+                  int num_records)
 {
     // We do a simple check on the marker count.
     // XXX: To test precisely skipping the instrs and data we'll need to spend
     // more effort here, but the initial delayed markers are the corner cases.
     int all_count = 0, marker_count = 0;
     for (const auto &memref : memrefs) {
-        if (all_count++ < skip_memrefs)
+        if (all_count++ < skip_records)
             continue;
-        if (all_count - skip_memrefs > num_memrefs)
+        if (all_count - skip_records > num_records)
             break;
         if (memref.marker.type == TRACE_TYPE_MARKER)
             ++marker_count;
     }
-    ASSERT(static_cast<size_t>(num_memrefs + skip_memrefs) <= memrefs.size(),
+    ASSERT(static_cast<size_t>(num_records + skip_records) <= memrefs.size(),
```

Comment:
The only reason we need to provide `memrefs` to test_skip_memrefs  is for this check? If there were too few memrefs in the provided `records` wouldn't it be detectable some other way too? Xref the other comment asking whether we can remove the duplication in run_limit_tests test setup.

Reply:
OK I refactored all 4 functions to use trace_entry_t, using readers to count memref types and boundaries.